src/Markdig/Syntax/ContainerBlock.cs

```diff
@@ -305,9 +306,9 @@ public BlockComparer(Comparison<Block> comparison)
                 this.comparison = comparison;
             }
 
-            public int Compare(Block x, Block y)
+            public int Compare(Block? x, Block? y)
             {
-                return comparison(x, y);
+                return comparison(x!, y!);
             }
```

Comment:
This should either accept non-nullable blocks, or the comparer should allow nullable

Reply:
This is an annoying case where the nullability contract of IComparer (below) forces us to define these as nullable.

``` int Compare([AllowNull] T x, [AllowNull] T y); ``` 

The System.Comparison delegate, sadly, does NOT define the system contract with nullable attributes. Our options are limited here since we don't control either implementation / contract.

Reply:
UPDATE: we can make the comparison explicitly accept nullable. But this forces us to suppress its use in Array.Sort -- and we don't actually have a case to compare against null. IComparer, regardless, requires us to accept null -- so I believe this is a valid case for suppression. We could also make this class oblivious.